pkg/build/builder/daemonless.go

```diff
@@ -306,10 +306,76 @@ func buildDaemonlessImage(sc types.SystemContext, store storage.Store, isolation
 func generateTransientMounts() []string {
 	mounts := []string{}
 	mounts = appendRHSMMount(mounts)
+	mounts = appendETCPKIMount(mounts)
+	mounts = appendRHRepoMount(mounts)
 	mounts = appendCATrustMount(mounts)
 	return mounts
 }
 
+func appendRHRepoMount(mounts []string) []string {
+	st, err := os.Stat("/run/secrets/redhat.repo")
+	if err != nil {
+		// since the presence of /run/secrets/redhat.repo is not a given, we won't log this a V(0) like
+		// we do for /run/secrets/rhsm
+		log.V(4).Infof("Failed to stat /run/secrets/redhat.repo, falling back to the Red Hat yum repository configuration in the build's base image. Error: %v", err)
+		return mounts
+	}
+	if st.IsDir() {
+		// if the file is there, but an unexpected type, then always have log show up via V(0)
+		log.V(4).Infof("Falling back to the Red Hat yum repository configuration in the build's base image: redhat.repo secrets location /run/secrets/redhat.repo is a directory.")
+		return mounts
+	}
+
+	// Add a bind of /run/secrets/redhat.repo, to pass along anything that the
+	// runtime mounted from the node into our /run/secrets/redhat.repo.
+	log.V(0).Infof("Adding transient rw bind mount for /run/secrets/redhat.repo")
+	tmpDir, err := ioutil.TempDir("/tmp", "redhat.repo-copy")
+	if err != nil {
+		log.V(4).Infof("Falling back to the Red Hat yum repository configuration in the base image: failed to create tmpdir for redhat.repo secret: %v", err)
```

Comment:
will switch to V0 per https://github.com/openshift/builder/pull/228#issuecomment-805110570 @adambkaplan